homeassistant/components/netgear/router.py

```diff
@@ -0,0 +1,286 @@
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_api(
+    password: str,
+    host: str = None,
+    username: str = None,
+    port: int = None,
+    ssl: bool = False,
+) -> Netgear:
+    """Get the Netgear API and login to it."""
+    api: Netgear = Netgear(password, host, username, port, ssl)
+
+    if not api.login():
+        raise CannotLoginException
+
+    return api
+
+
+@callback
+def async_setup_netgear_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    entity_class_generator: Callable[["NetgearRouter", dict], list],
+) -> None:
+    """Set up device tracker for Netgear component."""
+    router = hass.data[DOMAIN][entry.unique_id]
+    tracked = set()
+
+    @callback
+    def _async_router_updated():
+        """Update the values of the router."""
+        async_add_new_entities(
+            router, async_add_entities, tracked, entity_class_generator
+        )
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, router.signal_device_new, _async_router_updated)
+    )
+
+    _async_router_updated()
+
+
+@callback
+def async_add_new_entities(router, async_add_entities, tracked, entity_class_generator):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
+
+        new_tracked.extend(entity_class_generator(router, device))
+        tracked.add(mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked, True)
+
+
+class NetgearRouter:
+    """Representation of a Netgear router."""
+
+    def __init__(self, hass: HomeAssistantType, entry: ConfigEntry) -> None:
+        """Initialize a Netgear router."""
+        self.hass = hass
+        self.entry = entry
+        self.entry_id = entry.entry_id
+        self.unique_id = entry.unique_id
+        self._host = entry.data.get(CONF_HOST)
+        self._port = entry.data.get(CONF_PORT)
+        self._ssl = entry.data.get(CONF_SSL)
+        self._username = entry.data.get(CONF_USERNAME)
+        self._password = entry.data[CONF_PASSWORD]
+
+        self._info = None
+        self.model = None
+        self.device_name = None
+        self.firmware_version = None
+
+        self._method_version = 1
+        consider_home_int = entry.options.get(
+            CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME.total_seconds()
+        )
+        self._consider_home = timedelta(seconds=consider_home_int)
+
+        self._api: Netgear = None
+        self._attrs = {}
+
+        self.devices = {}
+
+    def _setup(self) -> None:
+        """Set up a Netgear router sync portion."""
+        self._api = get_api(
+            self._password,
+            self._host,
+            self._username,
+            self._port,
+            self._ssl,
+        )
+
+        self._info = self._api.get_info()
+        self.device_name = self._info["DeviceName"]
```

Comment:
Same problem here:
```
2021-09-13 14:05:19 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry R6300v2 for netgear
Traceback (most recent call last):
  File "/home/martin/dev/home-assistant/core/homeassistant/config_entries.py", line 304, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/home/martin/dev/home-assistant/core/homeassistant/components/netgear/__init__.py", line 21, in async_setup_entry
    await router.async_setup()
  File "/home/martin/dev/home-assistant/core/homeassistant/components/netgear/router.py", line 146, in async_setup
    await self.hass.async_add_executor_job(self._setup)
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/martin/dev/home-assistant/core/homeassistant/components/netgear/router.py", line 137, in _setup
    self.device_name = self._info["DeviceName"]
KeyError: 'DeviceName'
```

Reply:
Sorry, fixed it there too with the latest commit (also search for other occurences of "DeviceName").
Could you try again?